src/components/tooltip/tooltip.js

```diff
@@ -63,6 +63,15 @@ function MdTooltipDirective($timeout, $window, $$rAF, $document, $mdUtil, $mdThe
         tooltipParent = angular.element(current || document.body),
         debouncedOnResize = $$rAF.throttle(function () { if (scope.visible) positionTooltip(); });
 
+    // Observing for tooltip direction changes
+    attr.$observe('mdDirection', function (val) {
```

Comment:
For consistency, I think we should `$watch( )` md-direction just like we watch `md-visible`.